File: weekOfOct2/majorityElement.py
class Solution(object):
    def majorityElement(self, nums):
        """
        :type nums: List[int]
        :rtype: List[int]
        """

        n = len(nums)
        limit = n/3

        majors = []
        counts = {}

        for num in nums:
            if num not in counts:
                counts[num] = 1
            else:
                counts[num] += 1

            if counts[num] > limit and num not in majors:
                majors.append(num)

            if len(majors) == 3:
                break

        return majors

File: weekOfOct2/majorityElement.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var majorityElement = function(nums) {
  const n = nums.length;
  const limit = n/3;

  let majors = [];
  const map = {};

  for (let i = 0; i < n; i++) {
      map[nums[i]] === undefined ? map[nums[i]] = 1 : map[nums[i]]++;
      console.log(map[nums[i]], limit)
      if (map[nums[i]] > limit && !majors.includes(nums[i])) majors.push(nums[i])
      if (majors.length === 3) break;
  }

  console.log(map)

  return majors;
};
